packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -215,6 +215,10 @@ class _MaterialScrollbarState extends RawScrollbarState<_MaterialScrollbar> {
   @override
   bool get enableGestures => widget.interactive ?? _scrollbarTheme.interactive ?? !_useAndroidScrollbar;
 
+  MaterialStateProperty<bool> get _trackVisibility => MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+    return _scrollbarTheme.trackVisibility?.resolve(states) ?? false;
+  });
+
   bool get _showTrackOnHover => widget.showTrackOnHover ?? _scrollbarTheme.showTrackOnHover ?? false;
```

Comment:
This could be removed and refactored out in favor of `_trackVisibility` above, and `Scrollbar.showTrackOnHover` can be deprecated.

Just like we wouldn't want to enumerate properties like thumbColor, thumbHoverColor, thumbDragColor.. etc. This should consolidate the states into one property to keep the API clean and easy to understand. :)